src/Pomelo.EntityFrameworkCore.MySql/Extensions/MySqlDbContextOptionsExtensions.cs

```diff
@@ -45,13 +45,22 @@ public static class MySqlDbContextOptionsExtensions
             Check.NotNull(optionsBuilder, nameof(optionsBuilder));
             Check.NotNull(connection, nameof(connection));
 
-            var csb = new MySqlConnectionStringBuilder(connection.ConnectionString)
-            {
-	            AllowUserVariables = true,
-	            UseAffectedRows = false
-            };
-
-            connection.ConnectionString = csb.ConnectionString;
+            var csb = new MySqlConnectionStringBuilder(connection.ConnectionString);
+	        if (csb.AllowUserVariables != true || csb.UseAffectedRows != false)
+	        {
+	            try
```

Comment:
I switched this to `try/catch` because even if a connection is Closed, setting the connection string will still fail if the connection has previously been opened.